### DISCUSSION TITLE
Afficher dynamiquement la date de dernière mise à jour d'une instance Lodex

### DISCUSSION THREAD

**Discussion by eonm-pro** [Inist-CNRS/lodex#2086] [Ideas]
En tant qu'utilisateur, je souhaiterais pouvoir afficher dynamiquement la date de dernière mise à jour d'une instance.

De la même manière qu'il existe une routine pour récupérer le nombre total de documents dans une instance, serait-il possible d'avoir une routine qui récupère la date de modification du `lodexStamp` la plus récente ?

**Comment by yannguillemin** [comment]
Je ne sais pas s'il y a réellement besoin d'une routine spécifique pour cela, car cela semble pouvoir être géré en JS en formatant le champ d'affichage sous la forme "Text -JavaScript/HTML templating". En attendant, même si ce n'est pas très élégant, je peux te proposer la solution suivante en passant par un enrichissement qui va reformater les dates dans une nouvelle colonne sous la forme de chaînes de caractères que l'on peut trier plus facilement.
<img width="838" alt="image" src="https://github.com/user-attachments/assets/172346f2-7e9d-47ee-b1a2-d39a56173c23">

Ensuite, il suffit de créer une ressource qui exécute la routine "distinct-by" sur une ressource basée sur les valeurs de la colonne nouvellement créée, de définir l'affichage sous la forme "Text - JavaScript/HTML templating", de trier par "Label" en ordre descendant et de définir le nombre max de champs à afficher à 1, puis d'adapter un peu la deuxième partie du code préenregistré pour enlever les légendes et les valeurs du tableau créé (cf code ci-dessous).
<img width="398" alt="image" src="https://github.com/user-attachments/assets/9875840c-c2b3-4fe2-8991-ad44375b1299">
<% /* On vérifie si les données sont une liste de paires _id / value */ %>
<% /* We check if the data is a list of _id / value pairs */ %>
<% if (Array.isArray(root.values) && root.values[0]._id !== undefined) { %>

    <% /* affichage des données sous forme de tableau */ %>
    <% /* display data in table format */ %>
    <table>
        <tbody>
        <% /* on crée une entrée dans le tableau pour chaque valeur */ %>
        <% /* create an entry in the table for each value */ %>
        <% for (let datum of root.values) { %>
                    <%= datum._id %>
        <% } %>
        </tbody>
    </table>
Et on obtient par ex :
![image](https://github.com/user-attachments/assets/8c8027ca-325e-4912-9303-854c57702e51)

**Reply by AnaelKremer** [reply]
je n'avais pas vu ce message, tu appliques cet enrichissement (loader ou dans lodex)
`path = lodexStamp.importedDate
value = fix(new Date()).thru(d => new Intl.DateTimeFormat('fr-FR', { year: 'numeric', month: 'long', day: 'numeric' }).format(d))`

Pour avoir la date en Français, et ensuite comme l'a dit yann, distinct-by puis format EJS pour l'afficher

**Comment by touv** [comment]
L'ajout d'une routine /api/run/dates pourraient donner les dates :

- de la publication
- du dernier import
